### PR TITLE
ci: cleanup

### PR DESCRIPTION
**Pull request by umarcor**
Let's try to deal with #622 in smaller and easier to review/debug chunks.

This PR is styling only. No feature changes.

### CODE REVIEW

File: .github/workflows/verible-ci.yml [hunk 1]
```diff
@@ -23,18 +24,19 @@ jobs:
...
-    env: ${{ matrix.env}}
-    name: Check · ${{ matrix.env.MODE}}
-
+        mode:
+        - test
+        - compile
+        - clean
+    env:
+      MODE: ${{ matrix.mode}}
+    name: Check · ${{ matrix.mode}}
```
Comment (mithro) [inline .github/workflows/verible-ci.yml:39]:
```suggestion
    name: Check · ${{ matrix.mode }}
```

File: .github/workflows/verible-ci.yml [hunk 2]
```diff
@@ -44,30 +46,30 @@ jobs:
...
         path: "/home/runner/.cache/bazel"
-        key: bazel-${{ env.MODE }}
+        key: bazel-${{ matrix.mode }}
 
-    - name: Install Dependencies
+    - name: Before Install
       run: ./.github/workflows/install.sh
 
-    - name: ${{ matrix.env.mode }} Verible
+    - name: Main script
```
Comment (mithro) [inline .github/workflows/verible-ci.yml:57]: I think the old name here was better?

File: .github/workflows/verible-ci.yml [hunk 3]
```diff
@@ -44,30 +46,30 @@ jobs:
...
       uses: actions/cache@v1
-      if: ${{ env.MODE != 'clean' }}
+      if: matrix.mode != 'clean'
       with:
         path: "/home/runner/.cache/bazel"
-        key: bazel-${{ env.MODE }}
+        key: bazel-${{ matrix.mode }}
 
-    - name: Install Dependencies
+    - name: Before Install
```
Comment (mithro) [inline .github/workflows/verible-ci.yml:54]: I think the old name here was better?

**Review by mithro — APPROVED** [review]
Couple of minor comments.

File: .github/workflows/verible-ci.yml (hunk 1 above)
Comment (umarcor) [reply .github/workflows/verible-ci.yml]: Done.

File: .github/workflows/verible-ci.yml (hunk 2 above)
Comment (umarcor) [reply .github/workflows/verible-ci.yml]: Done.

File: .github/workflows/verible-ci.yml (hunk 3 above)
Comment (umarcor) [reply .github/workflows/verible-ci.yml]: Done.

**Comment by hzeller** [comment]
Ready to merge @mithro ?

**Review by mithro — APPROVED** [review]
LGTM!

**Pull request merged**